candidatebot.py

```diff
@@ -17,9 +18,44 @@
 XML_FILE = "CandidateSummaryAction.xml"
 # Limit what this does during testing.
 MAX_PAGES_TO_CREATE = 3
+# Rate-limit to five page-creations every second. Change to 0.1 before running
+# against Wikipedia, but it's fine to hammer cso.noidea.dog.
+EDIT_PAGES_PER_SECOND = 4
+QUERY_PAGES_PER_SECOND = 20
 
 # TODO: Set a user agent.
 
+def rate_limited(max_per_second):
+  """Rate limiting decorator-with-args.
+  Args:
+    max_per_second: (float) How many times per second to do the thing.
+  Returns:
+    (func): A decorator
+  """
+  interval = 1.0 / float(max_per_second)
+
+  def decorator(func):
```

Comment:
Meh - nevermind - I didn't notice the indentation to see that this is an internal function